### PR TITLE
Add perpendicular baseline function

### PR DESCRIPTION
**Pull request by rtburns-jpl**
Note that since isce3 is not required at toplevel by any other modules, this module is not imported by default. It must be explicitly imported via e.g. `from dolphin import baseline`, pulling in the isce3 dependency.

### CODE REVIEW

**Comment by scottstanie** [comment]
https://github.com/rtburns-jpl/dolphin/compare/baseline...scottstanie:dolphin:baseline-grid?expand=1 Thanks for getting the math working 👍  I was playing around with the old gist that I had to make a 3d cube with this.

I'm not actually sure if we want all of the OPERA parsing that is in my test branch though. @hfattahi @mirzaees We we thinking of just having the high-level call to isce3 here in dolphin, then the metadata cube in disp-s1? Or have a cube-making function here too?

**Comment by mirzaees** [comment]
@scottstanie I was picturing two use cases for orbits, one for network unwrapping which really requires one value rather than a cube, and the other is dem error if we want to add in the future  
I don't have an strong idea but do you think dolphin standalone should be able to access the baselines cube for these functionalities?

**Comment by scottstanie** [comment]
Ok @rtburns-jpl [opera-utils](https://github.com/opera-adt/opera-utils/releases) now has all the CSLC parsing functions, so the only remaining relevant commants are the refactors/doc strings updates to `compute` and the grid function[^1].

[^1]: I'm seeing that I called it a "baseline cube", even though I removed the 3rd loop over heights. it should probably be "grid" since it's 2D

File: src/dolphin/baseline.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    # Difference in position between the two passes
+    baseline = np.linalg.norm(sec_pos - ref_pos)
+
+    # Compute angle between LOS vector and baseline vector
+    # via the law of cosines
+    costheta = (ref_rng**2 + baseline**2 - sec_rng**2) / (2 * ref_rng * baseline)
+
+    sintheta = np.sqrt(1 - costheta**2)
+    perp = baseline * sintheta
+    # par = baseline * costheta
```
Comment (scottstanie) [inline src/dolphin/baseline.py:56]:
```suggestion
```

File: src/dolphin/baseline.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+        pos_ref, velocity = orbit_ref.interpolate(az_time_ref)
+        pos_sec, _ = orbit_sec.interpolate(az_time_sec)
+        b = compute(
+            llh_rad, pos_ref, pos_sec, range_ref, range_sec, velocity, ellipsoid
+        )
+
+        baselines.append(b)
+
+    baseline_cube = np.array(baselines).reshape(lon_grid.shape)
```
Comment (scottstanie) [inline src/dolphin/baseline.py:155]:
```suggestion
    baseline_grid = np.array(baselines).reshape(lon_grid.shape)
```

File: src/dolphin/baseline.py [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+    llh : ArrayLike
+        Lon/Lat/Height vector specifying the target position.
+    ref_pos : ArrayLike
+        Reference position vector (x, y, z) in ECEF coordinates.
+    sec_pos : ArrayLike
+        Secondary position vector (x, y, z) in ECEF coordinates.
+    ref_rng : float
+        Range from the reference satellite to the target.
+    sec_rng : float
+        Range from the secondary satellite to the target.
```
Comment (scottstanie) [inline src/dolphin/baseline.py:35]:
```suggestion
    ref_range : float
        Range from the reference satellite to the target.
    sec_range : float
        Range from the secondary satellite to the target.
```

File: src/dolphin/baseline.py [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+    """
+    # Difference in position between the two passes
+    baseline = np.linalg.norm(sec_pos - ref_pos)
+
+    # Compute angle between LOS vector and baseline vector
+    # via the law of cosines
+    costheta = (ref_rng**2 + baseline**2 - sec_rng**2) / (2 * ref_rng * baseline)
+
+    sintheta = np.sqrt(1 - costheta**2)
+    perp = baseline * sintheta
```
Comment (scottstanie) [inline src/dolphin/baseline.py:55]:
```suggestion
    cos_theta = (ref_range**2 + baseline**2 - sec_range**2) / (2 * ref_range * baseline)

    sin_theta = np.sqrt(1 - cos_theta**2)
    perp = baseline * sin_theta
```

File: src/dolphin/baseline.py [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+    # par = baseline * costheta
+
+    targ_xyz = ell.lon_lat_to_xyz(llh)
```
Comment (scottstanie) [inline src/dolphin/baseline.py:58]:
```suggestion
    target_xyz = ell.lon_lat_to_xyz(llh)
```

File: src/dolphin/baseline.py [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+    direction = np.sign(
+        np.dot(np.cross(targ_xyz - ref_pos, sec_pos - ref_pos), ref_vel)
```
Comment (scottstanie) [inline src/dolphin/baseline.py:60]:
```suggestion
        np.dot(np.cross(target_xyz - ref_pos, sec_pos - ref_pos), ref_vel)
```

**Review by scottstanie — APPROVED** [review]
👍 Just a few naming changes, I realized that I like spelling out `range` because [`rng` has started to mean random number generator](https://numpy.org/doc/stable/reference/random/generator.html) more than "range" to me

File: src/dolphin/baseline.py (hunk 1 above)
Comment (rtburns-jpl) [reply src/dolphin/baseline.py:56]: I like this as a demonstration of what the other trig component gives, even though the result isn't used here. If I wasn't familiar with the algorithm my first question would be what the cos-theta component would be.

Comment (scottstanie) [reply src/dolphin/baseline.py]: Ok gotcha, that's true that it's helpful for that.
How about if we make ie...
```  python
    # parallel_baseline = baseline * cosine_theta

```
 (I updated the comment above)

**Pull request merged**